system/updated/updated.py

```diff
@@ -0,0 +1,246 @@
+  return is_git_repo(current_directory) or \
+         build_metadata.channel != other_metadata.channel or \
+         build_metadata.openpilot.git_commit != other_metadata.openpilot.git_commit or \
+         build_metadata.openpilot.build_style != other_metadata.openpilot.build_style
+
+
+def create_remote_chunk_source(caibx_url, chunks):
+  return ('remote', casync.RemoteChunkReader(casync.get_default_store(caibx_url)), casync.build_chunk_dict(chunks))
+
+
+def get_partition_hash(path: str, partition_size: int, full_check: bool) -> str:
+  if full_check:
+    hasher = hashlib.sha256()
+    pos, chunk_size = 0, 1024 * 1024
+
+    with open(path, 'rb') as out:
+      while pos < partition_size:
+        n = min(chunk_size, partition_size - pos)
+        hasher.update(out.read(n))
+        pos += n
+
+    return hasher.hexdigest().lower()
+  else:
+    with open(path, 'rb+') as out:
+      out.seek(partition_size)
+      return out.read(64).decode("utf-8")
+
+
+def set_partition_hash(path: str, partition_size: int, new_hash: bytes):
+  with open(path, 'rb+') as out:
+    out.seek(partition_size)
+    assert len(new_hash) == 64
+    out.write(new_hash)
+
+  os.sync()
+
+
+def extract_directory_helper(entry, cache_directory, directory):
+  caibx_url = entry["casync"]["caibx"]
+  target = casync.parse_caibx(caibx_url)
+
+  shutil.rmtree(directory)
+
+  cache_filename = os.path.join(CASYNC_TMPDIR, "cache.tar")
+  tmp_filename = os.path.join(CASYNC_TMPDIR, "tmp.tar")
+
+  cloudlog.info("building tarball update cache...")
+  start = time.monotonic()
+  sources = [('cache', casync.DirectoryTarChunkReader(cache_directory, cache_filename), casync.build_chunk_dict(target))]
+  cloudlog.info(f"tarball cache creation completed in {time.monotonic() - start} seconds")
+
+  sources += [create_remote_chunk_source(caibx_url, target)]
+
+  cloudlog.info(f"extracting {caibx_url} to {directory}")
+  start = time.monotonic()
+  stats = casync.extract_directory(target, sources, directory, tmp_filename)
+  cloudlog.info(f"extraction completed in {time.monotonic() - start} seconds with {stats}")
+
+
+def extract_partition_helper(entry):
+  cloudlog.info(f"extracting partition: {entry}")
+
+  caibx_url = entry["casync"]["caibx"]
+  chunks = casync.parse_caibx(caibx_url)
+
+  target_path = entry["path"]
+
+  sources = []
+
+  if entry["ab"]:
+    target_path = entry["path"].replace("_a", HARDWARE.get_target_ab_slot())
+    seed_path = entry["path"].replace("_a", HARDWARE.get_current_ab_slot())
+    sources += [('seed', casync.FileChunkReader(seed_path), casync.build_chunk_dict(chunks))]
+
+  sources += [
+    ('target', casync.FileChunkReader(target_path), casync.build_chunk_dict(chunks)),
+    create_remote_chunk_source(caibx_url, chunks)
+  ]
+
+  current_hash = get_partition_hash(target_path, entry["size"], entry["full_check"])
+  target_hash = entry['hash_raw'].lower()
+
+  cloudlog.info(f"{current_hash=}, {target_hash=}")
+  if current_hash == target_hash:
```

Comment:
let's add an environment variable to always write. when we first enable the new updater in master, I want to hit all the code paths. we won't leave it like that just to reduce wear on the flash